### DISCUSSION TITLE
Support for Proxy ARP in Foomuuri

### DISCUSSION THREAD

**Discussion by realsimix** [FoobarOy/foomuuri#2] [Ideas]
Hi,

We are running Shorewall on our firewalls for many years now. It has served us very well in the past but I understand the day will come where we'll need to have a look at new solutions. Now that Foomuuri was announced on the Shorewall list I'm interested to learn what it does provide.

While studying Foomuuri I found one thing which is not supported at the moment: setting up Proxy ARP to build a DMZ without the need for NAT. We're making heavy use of the feature present in Shorewall https://shorewall.org/ProxyARP.htm and I'm wondering if it would be possible to add something similar to Foomuuri?

Thanks and regards,
Simon

**Comment by kimheino** [comment]
Hi,

Short answer is that unfortunately we think that configuring proxy ARP falls out of Foomuuri's scope. Correct me if I'm wrong, see the long answer.

Much longer answer is that Foomuuri's main target is to configure packet filtering rules. Things that can be done with NetworkManager, static sysctl, static `ip` commands, etc. are left to be configured with those tools. Shorewall has lot of features, some are filtering rules, some are more-or-less helpers to configure static network settings. We try to not add too much such helpers to avoid bloating Foomuuri too much. As far as I know, proxy ARP can be configured with `ip neigh add proxy 192.168.1.100 dev eth0` commands. Simply running these commands is enough, Foomuuri doesn't need to know anything about them.

Your question gave me an idea to add pre/post start/stop hooks to Foomuuri. So basically soon you can do in Foomuuri:

```
hook {
  post_start /etc/foomuuri/setup_proxyarp.sh
}
```

And then run `ip` commands in `setup_proxyarp.sh`. Would this be enough for your needs?

**Comment by realsimix** [comment]
Thanks, having hooks for such things seems a good idea. Not only for proxy ARP but also for other things.
